styx-api-service/src/main/java/com/spotify/styx/api/deprecated/CliResource.java

```diff
@@ -114,21 +115,18 @@ public CliResource(StatusResource statusResource, String schedulerServiceBaseUrl
     } catch (JsonProcessingException e) {
       throw Throwables.propagate(e);
     }
-    final Request request =
-        rc.request().withUri(schedulerServiceBaseUrl + SCHEDULER_BASE_PATH + TRIGGER_PATH)
-            .withPayload(payload);
-    return rc.requestScopedClient().send(request).thenApply(response -> {
-      final com.spotify.styx.model.WorkflowInstance responsePayload;
-      try {
-        responsePayload = Json.deserialize(response.payload().get(),
-                             com.spotify.styx.model.WorkflowInstance.class);
-      } catch (IOException e) {
-        throw Throwables.propagate(e);
-      }
-      return Response.forStatus(response.status())
-          .withPayload(WorkflowInstance.create(responsePayload))
-          .withHeaders(response.headers());
-    });
+    final Request request = rc.request()
+        .withUri(schedulerServiceBaseUrl + SCHEDULER_BASE_PATH + TRIGGER_PATH)
+        .withPayload(payload);
+    return rc.requestScopedClient().send(request).thenApply(response ->
+        response.withPayload(response.payload().map(p -> {
+          try {
+            return WorkflowInstance.create(
+                Json.deserialize(p, com.spotify.styx.model.WorkflowInstance.class));
+          } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
` throw Throwables.propagate(e);` seems to be used everywhere and feels more reliable.